src/main/scala/beam/agentsim/infrastructure/power/PowerController.scala

```diff
@@ -62,17 +64,52 @@ class PowerController(chargingNetworkMap: Map[Id[VehicleManager], ChargingNetwor
       physicalBounds = beamFederateOption match {
         case Some(beamFederate) if helicsConfig.connectionEnabled && estimatedLoad.isDefined =>
           logger.debug("Sending power over next planning horizon to the grid at time {}...", currentTime)
-          beamFederate.publishJSON(
-            estimatedLoad.get.map {
-              case (station, powerInKW) =>
-                from(station.zone) ++ Map("estimatedLoad" -> powerInKW)
-            }.toList
-          )
-          val (_, gridBounds) = beamFederate.syncAndCollectJSON(currentTime)
+          // PUBLISH
+          val msgToPublish = estimatedLoad.get.map {
+            case (station, powerInKW) =>
+              Map(
+                "managerId"         -> station.zone.managerId,
+                "tazId"             -> station.zone.tazId.toString,
+                "parkingType"       -> station.zone.parkingType.toString,
+                "chargingPointType" -> station.zone.chargingPointType.toString,
+                "numChargers"       -> station.zone.numChargers,
+                "estimatedLoad"     -> powerInKW
+              )
+          }
+          beamFederate.publishJSON(msgToPublish.toList)
+
+          var gridBounds = List.empty[Map[String, Any]]
+          while (gridBounds.isEmpty) {
+            // SYNC
+            beamFederate.sync(currentTime)
+            // COLLECT
+            gridBounds = beamFederate.collectJSON()
+          }
```

Comment:
would it be better to insert thread.sleep at the end of the loop body to decrease CPU consumption?

Reply:
I added the thread sleep